appinventor/appengine/src/com/google/appinventor/client/TopToolbar.java

```diff
@@ -180,16 +180,31 @@ public TopToolbar() {
 
     initWidget(toolbar);
   }
+  
+  public void updateMoveToTrash(int flag){
+	if(flag == 1){
+	   flag = 0;
+       DropDownItem item = new DropDownItem(WIDGET_NAME_DELETE,MESSAGES.trashProjectMenuItem(),  new DeleteForeverProjectAction());
+       fileDropDown.setItemEnabled(MESSAGES.trashProjectMenuItem(), true);
+       fileDropDown.replaceSpecificItem(item, 5);
+       
+    }
+    else{
+	   DropDownItem item = new DropDownItem(WIDGET_NAME_MY_PROJECTS, MESSAGES.projectMenuItem(), new DeleteAction());
+	   fileDropDown.replaceSpecificItem(item, 6);
+    }
+  } 
 
   public void updateMenuState(int numSelectedProjects, int numProjects) {
     boolean allowDelete = !isReadOnly && numSelectedProjects > 0;
     boolean allowExport = numSelectedProjects > 0;
     boolean allowExportAll = numProjects > 0;
-    fileDropDown.setItemEnabled(MESSAGES.trashProjectMenuItem(), allowDelete);
+    fileDropDown.setItemEnabled(MESSAGES.trashProjectMenuItem(), true);
```

Comment:
Did you intend to remove the allowDelete check here? AllowDelete is set to true only if the user has permissions to trash projects, which is not always the case.